cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -346,7 +347,13 @@ func GetConfigMapWithShortRetry(client clientset.Interface, namespace, name stri
 			if err == nil {
 				return true, nil
 			}
-			lastError = err
+			// If some code is about to go over the context deadline, "x/time/rate/rate.go" would return
+			// and untyped error with the string "would exceed context deadline". If some code already exceeded
+			// the deadline the error would be of type DeadlineExceeded. Ignore such context errors and only store
+			// API and connectivity errors.
```

Comment:
It sounds like that the error may be related to `context.Background()`.

This can fix the current CI failure. Do we have other potential error messages to handle now or may have in the future?

Reply:
> It sounds like that the error may be related to context.Background().

it's a general problem for any context wtih a deadline. other places in k8s can also hit such a problem, when the `Poll[Immediate]` deprecated functions are replaced. it depends on what error is expected when a poll completes.

when we started to use PollUntilContextTimeout we started seeing this "would exceed context deadline" error as the "last error" we get, but actually we anticipated our "last error" to get to be an API error when a resource is missing.

> This can fix the current CI failure. Do we have other potential error messages to handle now or may have in the future?

i don't think so, these are the 2 relevant errors that can happen if a context with a deadline is passed to a go client.

Reply:
Got it.